### PR TITLE
absolute layout docs

### PR DESCRIPTION
**Pull request by danielftz**
This is the corresponding documentation for pull request here: https://github.com/CommunityToolkit/Maui.Markup/pull/31

### CODE REVIEW

**Comment by PRMerger17** [comment]
@danielftz : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger5** [comment]
@danielftz : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger-2** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 File Change Percent | This PR contains file(s) with more than 20% file change.

File: docs/maui/markup/markup.md [hunk 1]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+
+C# Markup includes a series of layout extension methods that support positioning views in layouts, and content in views:
+
+| Type | Extension methods |
+| ----------- | ----------- |
+| `AbsoluteLayout` | `LayoutFlags`, `LayoutBounds` |
+
+#AbsoluteLayout LayoutBounds
+
+One or a combination of `double`, `Point` `Size` and `Rect` can be used to define the `LayoutBounds` property of [AbsoluteLayout](https://docs.microsoft.com/dotnet/maui/user-interface/layouts/absolutelayout). The following code shows an example of how to define and consume AbsoluteLayout markup extension methods.
```
Comment (bijington) [inline docs/maui/markup/markup.md:136]: Links to docs site pages should be relative.

```suggestion
One or a combination of `double`, `Point` `Size` and `Rect` can be used to define the `LayoutBounds` property of [AbsoluteLayout](/dotnet/maui/user-interface/layouts/absolutelayout). The following code shows an example of how to define and consume AbsoluteLayout markup extension methods. 
```

File: docs/maui/markup/markup.md [hunk 2]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+        {
+            Content = new AbsoluteLayout
+            {
+                Children =
+                {
+                    new BoxView
+                    {
+                        Color = Colors.Blue,
+                    }.LayoutFlags(AbsoluteLayoutFlags.PositionProportional)
+                    .LayoutBounds(0.5,0,100,25),
```
Comment (bijington) [inline docs/maui/markup/markup.md:154]: Spacing should be consistent with coding guidelines

```suggestion
                    .LayoutBounds(0.5, 0, 100, 25),
```

File: docs/maui/markup/markup.md [hunk 3]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+
+                    new BoxView
+                    {
+                        Color = Colors.Green,
+                        WidthRequest = 25,
+                        HeightRequest = 100,
+                    }.LayoutFlags(AbsoluteLayoutFlags.PositionProportional)
+                    .LayoutBounds(0,0.5),
```
Comment (bijington) [inline docs/maui/markup/markup.md:162]: Spacing should be consistent with coding guidelines

```suggestion
                    .LayoutBounds(0, 0.5),
```

File: docs/maui/markup/markup.md [hunk 4]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+
+                    new BoxView
+                    {
+                        Color = Colors.Red,
+                        WidthRequest = 25,
+                        HeightRequest = 100,
+                    }.LayoutFlags(AbsoluteLayoutFlags.PositionProportional)
+                    .LayoutBounds(new Point(1,0.5)),
```
Comment (bijington) [inline docs/maui/markup/markup.md:170]: Spacing should be consistent with coding guidelines

```suggestion
                    .LayoutBounds(new Point(1, 0.5)),
```

File: docs/maui/markup/markup.md [hunk 5]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+
+                    new BoxView
+                    {
+                        Color = Colors.Grey,
+                    }.LayoutFlags(AbsoluteLayoutFlags.PositionProportional)
+                    .LayoutBounds(new Point(0.5,1), new Size(100,25)),
```
Comment (bijington) [inline docs/maui/markup/markup.md:176]: Spacing should be consistent with coding guidelines

```suggestion
                    .LayoutBounds(new Point(0.5, 1), new Size(100, 25)),
```

File: docs/maui/markup/markup.md [hunk 6]
```diff
@@ -120,4 +120,69 @@ C# Markup extensions also allow developers to use an `enum` to define names for
...
+
+                    new BoxView
+                    {
+                        Color = Colors.Tan,
+                    }.LayoutFlags(AbsoluteLayoutFlags.All)
+                    .LayoutBounds(new Rect(0.5,0.5,1d/3d, 1d/3d))
```
Comment (bijington) [inline docs/maui/markup/markup.md:182]: Spacing should be consistent with coding guidelines

```suggestion
                    .LayoutBounds(new Rect(0.5, 0.5, 1d/3d, 1d/3d))
```

**Review by bijington — CHANGES_REQUESTED** [review]
@danielftz thank you so much for the contribution, sorry I have submitted quite a few points, hopefully most are pretty minor. The big change will be to move this in to its own separate file and then we link to it from the `TOC.yml` file. Sorry I appreciate that we don't have any docs on this kind of approach yet but I will make it my mission to fix that this week. Would you feel comfortable with the restructuring or would you be happy for me to move things around?

**Comment by danielftz** [comment]
> @danielftz thank you so much for the contribution, sorry I have submitted quite a few points, hopefully most are pretty minor. The big change will be to move this in to its own separate file and then we link to it from the `TOC.yml` file. Sorry I appreciate that we don't have any docs on this kind of approach yet but I will make it my mission to fix that this week. Would you feel comfortable with the restructuring or would you be happy for me to move things around?

Please move things around for me if it is not too much trouble

**Comment by PRMerger3** [comment]
@jfversluis : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger14** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by jfversluis** [comment]
Something seems off in the final markup near the Layout heading

image

Additionally, I know it's not related to this change, somewhere near the start there is a "(view))" with double brackets at the end. I couldn't comment on the line, sorry 😅

**Comment by bijington** [comment]
> > @danielftz thank you so much for the contribution, sorry I have submitted quite a few points, hopefully most are pretty minor. The big change will be to move this in to its own separate file and then we link to it from the `TOC.yml` file. Sorry I appreciate that we don't have any docs on this kind of approach yet but I will make it my mission to fix that this week. Would you feel comfortable with the restructuring or would you be happy for me to move things around?
> 
> Please move things around for me if it is not too much trouble

Yes no problem I can look to get this sorted this week

**Comment by PRMerger15** [comment]
@danielftz : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger12** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger16** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 File Change Percent | This PR contains file(s) with more than 20% file change.

**Comment by PRMerger5** [comment]
@bijington : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by bijington** [comment]
@jfversluis is it possible to see how this now previews please?

**Comment by bijington** [comment]
perhaps the `Markup.md` file should contain a table linking to each of the extension pages?

**Comment by jfversluis** [comment]
It might appear a bit zoomed, I noticed I had the zoom on in the browser :) Seems to look good to me!

image

image

**Comment by bijington** [comment]
Oh great thanks! I wonder if we want to have a specific section in the TOC for extensions ->
Screenshot 2022-03-22 at 09 15 58

**Review by bijington — APPROVED** [review]

**Comment by jfversluis** [comment]
If it's good enough for Shaun, it's good enough for me!

Thanks so much @danielftz!

**Comment by bijington** [comment]
Yes thank you @danielftz we really appreciate the effort

**Pull request merged**